chimney/src/main/scala/io/scalaland/chimney/dsl/ConfigDsl.scala

```diff
@@ -1,8 +1,24 @@
 package io.scalaland.chimney.dsl
 
+import io.scalaland.chimney.TransformerFlags._
 import io.scalaland.chimney.internal.TransformerCfg
 import io.scalaland.chimney.internal.TransformerCfg._
 
+trait AConfigDsl[FlagsContainer[_], Flags] {
+  def enableDefaultValues: FlagsContainer[Flags with EnableDefaultValues] =
+    this.asInstanceOf[FlagsContainer[Flags with EnableDefaultValues]]
+
+  def disableDefaultValues: FlagsContainer[Flags with DisableDefaultValues] =
+    this.asInstanceOf[FlagsContainer[Flags with DisableDefaultValues]]
+
+  def enableUnsafeOption: FlagsContainer[Flags with EnableUnsafeOption] =
+    this.asInstanceOf[FlagsContainer[Flags with EnableUnsafeOption]]
+
+  def disableUnsafeOption: FlagsContainer[Flags with DisableUnsafeOption] =
```

Comment:
Separation of `EnableX` and `DisableX` allows for ambiguous representations like `EnableX with DisableX` - which one should be used?

It would be much better to have `X` only, so that we would test presence of `X` in the compound type in order to determine whether we want to enable the flag. But then operating on the flags type would be much more challenging:
- in trivial implementation, we can enable something twice, obtaining `X with X`, so perhaps best to check for X presence before adding to the compound type
- we shouldn't be able to disable something that is not enabled

Both can be solved either by some implicit type-level computation or by macros. But it would be very nice to keep them simple at the same time...